src/main/java/com/conveyal/datatools/manager/controllers/api/LabelController.java

```diff
@@ -0,0 +1,219 @@
+     */
+    private static Label updateLabel(Request req, Response res) throws IOException {
+        String labelId = req.params("id");
+        Label formerLabel = requestLabelById(req, Actions.MANAGE);
+        Label updatedLabel = getPOJOFromRequestBody(req, Label.class);
+
+        // Some things shouldn't be updated
+        updatedLabel.projectId = formerLabel.projectId;
+        updatedLabel.id = formerLabel.id;
+
+        validate(req, updatedLabel);
+
+        Persistence.labels.replace(labelId, updatedLabel);
+
+        return updatedLabel;
+    }
+    /**
+     * Spark HTTP endpoint to delete a label
+     *
+     * FIXME: Should this just set a "deleted" flag instead of removing from the database entirely?
+     */
+    private static Label deleteLabel(Request req, Response res) {
+        Label label = requestLabelById(req, Actions.MANAGE);
+
+        // More specific error message if label doesn't exist
+        if (label == null) {
+            logMessageAndHalt(req, 403, "Label does not exist.");
+            return null;
+        }
+
+        try {
+            // Iterate over feed sources and remove any references to the label
+            Persistence.feedSources.getAll().forEach(feedSource -> {
+                List<String> newLabels = feedSource.labels.stream()
+                        .filter(l -> !l.equals(label.id))
+                        .collect(Collectors.toList());
+                // If there are changes, update the feed source
+                if (!newLabels.equals(feedSource.labels)) {
+                    FeedSource fs = Persistence.feedSources.getById(feedSource.id);
+                    fs.labels = newLabels;
+                    Persistence.feedSources.replace(fs.id, fs);
+                }
+            });
+
+            label.delete();
+            return label;
+        } catch (Exception e) {
+            logMessageAndHalt(req, 500, "Unknown error occurred while deleting label.", e);
+            return null;
+        }
+    }
+
+    /**
+     * Helper function returns label if user has permission for specified action.
+     * @param req spark Request object from API request
+     * @param action action type (either "view" or Permission.MANAGE)
+     * @return label object for ID
+     */
+    private static Label requestLabelById(Request req, Actions action) {
+        String id = req.params("id");
+        if (id == null) {
+            logMessageAndHalt(req, 400, "Please specify id param");
+        }
+        Label label = Persistence.labels.getById(id);
+        checkLabelPermissions(req, label, action);
+
+        return label;
+    }
+
+    /** Helper method returns label only if user is authorized to do given action with it
+     *
+     * @param req       Spark request used for determining user permissions
+     * @param label     Label to check
+     * @param action    Action to be taken on label, changes who can do what
+     * @return          Label which is ok to return to user
+     */
```

Comment:
Please change JavaDoc to remove "returns" wording.

Reply:
Good catch thank you. Removed!